File: backend/backend/src/main/java/com/rentnest/backend/entity/Property.java
package com.rentnest.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "properties")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Property {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(length = 2000)
    private String description;

    @Column(nullable = false)
    private String location;

    private String address;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PropertyType propertyType;

    private Integer bedrooms;

    private Integer bathrooms;

    @Column(nullable = false)
    private BigDecimal price;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PropertyStatus status;

    private String imageUrl;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @PrePersist
    public void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();

        if (this.status == null) {
            this.status = PropertyStatus.AVAILABLE;
        }
    }

    @PreUpdate
    public void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/types/property.ts
export type PropertyStatus =
  | "AVAILABLE"
  | "UNDER_INSPECTION"
  | "RENTED"
  | "UNAVAILABLE";

export type PropertyType =
  | "APARTMENT"
  | "STUDIO"
  | "DUPLEX"
  | "BUNGALOW"
  | "SELF_CONTAIN"
  | "COMMERCIAL";

export interface Property {
  id: number;
  title: string;
  description: string;
  location: string;
  address: string;
  propertyType: PropertyType;
  bedrooms: number;
  bathrooms: number;
  price: number;
  status: PropertyStatus;
  imageUrl: string;
  createdAt: string;
  updatedAt: string;
}